### PR TITLE
Improve Python/C++ tutorial

### LINKED ISSUE #4271: DLL name for python addins
I just spent a frustrating few hours trying to get addins written in C++ to work in Python. What neither this page nor the equivalent page at https://docs.python.org mention is that the resulting dll must be renamed to &lt;modulename&gt;.pyd (you can of course do this in the linker with a /OUT directive). 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 40e3b6b1-a3d2-f4f2-b1ad-dd654a062343
* Version Independent ID: b2418b84-04f4-60d2-ec1d-291a5d4d0c51
* Content: [Write C++ extensions for Python - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/python/working-with-c-cpp-python-in-visual-studio?view=vs-2019#feedback)
* Content Source: [docs/python/working-with-c-cpp-python-in-visual-studio.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/python/working-with-c-cpp-python-in-visual-studio.md)
* Product: **visual-studio-windows**
* Technology: **vs-python**
* GitHub Login: @JoshuaPartlow
* Microsoft Alias: **joshuapa**

### LINKED ISSUE #6425: C1189: Py_LIMITED_API is incompatible with Py_DEBUG, Py_TRACE_REFS, and Py_REF_DEBUG still here
I've set Multi-threaded DLL (/MD) for all configurations but "C1189: Py_LIMITED_API is incompatible with Py_DEBUG, Py_TRACE_REFS, and Py_REF_DEBUG" error still happened until I removed _DEBUG preprocessor definition under debug configuration.

image

After that everything seems ok, mixed mode debugging works fine.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 40e3b6b1-a3d2-f4f2-b1ad-dd654a062343
* Version Independent ID: b2418b84-04f4-60d2-ec1d-291a5d4d0c51
* Content: [Write C++ extensions for Python - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/python/working-with-c-cpp-python-in-visual-studio?view=vs-2019)
* Content Source: [docs/python/working-with-c-cpp-python-in-visual-studio.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/python/working-with-c-cpp-python-in-visual-studio.md)
* Product: **visual-studio-windows**
* Technology: **vs-python**
* GitHub Login: @JoshuaPartlow
* Microsoft Alias: **joshuapa**

### PR DESCRIPTION
**Pull request by zooba**
Closes #4271 and closes #6425

### CODE REVIEW

**Comment by PRMerger6** [comment]
@zooba : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by zooba** [comment]
Just posting progress while I step away for a bit - still need to reread my own work before calling it ready, but feel free to chip in with comments if you have any.

**Comment by PRMerger18** [comment]
@zooba : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/python/working-with-c-cpp-python-in-visual-studio.md [hunk 1]
```diff
@@ -335,17 +328,27 @@ The alternate method, described in the following steps, installs the module in t
     )
     ```
 
-1. The *setup.py* code instructs Python to build the extension using the Visual Studio 2015 C++ toolset when used from the command line. Open an elevated command prompt, navigate to the folder containing the C++ project (that is, the folder that contains *setup.py*), and enter the following command:
+1. Create a second file named *pyproject.toml* in the C++ project, and paste the following code into it:
 
-    ```command
-    pip install .
+    ```toml
```
Comment (zooba) [inline docs/python/working-with-c-cpp-python-in-visual-studio.md:333]: Is `toml` a valid syntax for our docs?

**Pull request closed without merging**